### PR TITLE
[RA-1257]- New Patient Registrations report completed

### PR DESCRIPTION
**Pull request by JudeNiroshan**
Built-in reports for reference application GSoC project. This PR includes the work has been up to 3rd week.
Following work has been completed so far:
- Implement suggested reports through SQL definitions. (Week #1)
- Implement New Patient Registrations report through cohort definitions. (Week #3)
- Unit Test coverage for New Patient Registrations report. (Week #3)

- [x] mvn clean install
- [x] reports were available once the module has deployed.

### CODE REVIEW

File: omod/src/main/resources/config.xml [hunk 1]
```diff
@@ -54,6 +54,14 @@
...
 			org.openmrs.module.metadatamapping
 		</require_module>
+
+		<require_module version="${reportingVersion}">
+			org.openmrs.module.reporting
+		</require_module>
+
+		<require_module version="${metadatadeployVersion}">
+			org.openmrs.module.metadatadeploy
+		</require_module>
```
Comment (rkorytkowski) [inline omod/src/main/resources/config.xml]: Why is this required?

File: api/src/test/java/org/openmrs/module/referenceapplication/reporting/reports/NumberOfPatientRegistrationsTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	@Override
+	public EvaluationContext getEvaluationContext() {
+		EvaluationContext context = new EvaluationContext();
+		context.addParameterValue("startDate", DateUtil.getDateTime(2017,6,1));
+		context.addParameterValue("endDate", DateUtil.getDateTime(2017,6,16));
+		context.setBaseCohort(new Cohort("50, 51"));
+		return context;
+	}
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/module/referenceapplication/reporting/reports/NumberOfPatientRegistrationsTest.java]: Could we check in this test for an actual result, e.g. that there was 5 patient registered in that period...?

**Comment by rkorytkowski** [comment]
Nice progress! It seems the changes should actually go into https://github.com/openmrs/openmrs-module-referencemetadata module instead of https://github.com/openmrs/openmrs-module-referenceapplication. Could you please ask on talk, which is preferred going forward?

**Comment by rkorytkowski** [comment]
Is this PR created instead of https://github.com/openmrs/openmrs-module-referenceapplication/pull/41 ? Please do not create new PRs rather push to the same branch and your PR will be automatically updated.

File: omod/src/main/resources/config.xml (hunk 1 above)
Comment (JudeNiroshan) [inline omod/src/main/resources/config.xml]: Yes, without this particular, there is a runtime error when I deploy the referenceApplication module. I had a chat about this with Daniel on IRC. However I will try to run the module by removing this dependency. If it doesn't, I'll keep this.

File: api/src/test/java/org/openmrs/module/referenceapplication/reporting/reports/NumberOfPatientRegistrationsTest.java (hunk 2 above)
Comment (JudeNiroshan) [inline api/src/test/java/org/openmrs/module/referenceapplication/reporting/reports/NumberOfPatientRegistrationsTest.java]: Yes, that's a good point. I checked the unit tests in rwanda reports they haven't done it. Whether the report is actually displaying the correct records or not wasn't attested. I'll check on it more.

**Comment by JudeNiroshan** [comment]
Yes Rafal. As Darius mentioned in our talk thread for GSoC project; I'll make a comment on the thread about this. Thank you for the review. BTW; sorry that I wasn't aware about the automatic PR modifications.

File: omod/src/main/resources/config.xml (hunk 1 above)
Comment (JudeNiroshan) [inline omod/src/main/resources/config.xml]: Hi Rafal, we need this entry in order to load the metadatadeploy module at the time of referenceApplication module is initializing. It is explained in this wiki page. [https://wiki.openmrs.org/display/docs/Requiring+another+module+in+your+module](https://wiki.openmrs.org/display/docs/Requiring+another+module+in+your+module)

**Comment by JudeNiroshan** [comment]
Closing the PR as these changes decided to be moved to ReferenceMetaData module.
